.github/workflows/codeql.yml

```
@@ -1,17 +1,101 @@
+              - '**/*.ts'
+              - '**/*.cts'
+              - '**/*.mts'
+              - '**/*.vue'
+            python:
+              - '**/*.py'
+  codeql-java:
+    name: Analyze Java source files
+    runs-on: ubuntu-latest
+    needs: check-changes
+    
+    if: env.analyze-java && (github.ref_name == 'main' || needs.check-changes.outputs.java == 'true')
```

actionlint (1.7.4)

context "env" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)